cmd/avrogen/main.go

```diff
@@ -75,12 +75,20 @@ func realMain(args []string, out io.Writer) int {
 	}
 	formatted, err := format.Source(buf.Bytes())
 	if err != nil {
-		_, _ = fmt.Fprintf(out, "Error: could format code: %v\n", err)
+		_, _ = fmt.Fprintf(out, "Error: could not format code: %v\n", err)
 		return 3
 	}
-	if err = os.WriteFile(cfg.Out, formatted, 0o600); err != nil {
-		_, _ = fmt.Fprintf(out, "Error: could write file: %v\n", err)
-		return 4
+
+	if cfg.Out == "" {
+		if _, err = dumpout.Write(formatted); err != nil {
+			_, _ = fmt.Fprintf(out, "Error: could not write to stdout: %v\n", err)
+			return 4
+		}
+	} else {
+		if err = os.WriteFile(cfg.Out, formatted, 0o600); err != nil {
+			_, _ = fmt.Fprintf(out, "Error: could not write file: %v\n", err)
+			return 4
+		}
```

Comment:
What do you think about standardising this by switching file writing to `os.Open` style. It would get rid of the smelly else and make this simpler.

Reply:
Ah yes, was thinking on implementing it like that but wanted to minimise the amount of changes.
Given that you are okay with that I'll do it.

Reply:
Done in 381d8a4eb424dd786c62c105558349949808ebec
Let me know what do you think.

Reply:
Btw, I noticed that the file mode was previously being set to `0o600`.
With `os.Create`, the default file mode is `0o666` before umask, which is a better, more convenient default.

Was there any specific reason for the previous file mode?

Reply:
Likely to appease gosec. It is not a large problem either way.

Reply:
gosec seems happy :)
```
$ go run github.com/securego/gosec/v2/cmd/gosec@latest ./cmd/avrogen/
[gosec] 2023/09/05 21:44:14 Including rules: default
[gosec] 2023/09/05 21:44:14 Excluding rules: default
[gosec] 2023/09/05 21:44:14 Import directory: /home/hhromic/avro/cmd/avrogen
[gosec] 2023/09/05 21:44:14 Checking package: main
[gosec] 2023/09/05 21:44:14 Checking file: /home/hhromic/avro/cmd/avrogen/main.go
Results:


Summary:
  Gosec  : dev
  Files  : 1
  Lines  : 147
  Nosec  : 0
  Issues : 0
```